core/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -89,6 +91,12 @@
     private final boolean ignoreUnknownFields;
 
     /**
+     * A special purpose field parser that gets used when the provided matchFieldPrecidate accepts it
+     */
+    private FieldParser matchFieldParser;
+    private Predicate<String> matchFieldPredicate;
```

Comment:
I like the idea of the predicate. It could for instance be something like `registry.supports(name, Aggregation.class)`. This way we could also support ignoreUnknownFields for forward compatibility